crates/deltalake-core/src/operations/merge.rs

```diff
@@ -1200,6 +1083,24 @@ async fn execute(
     Ok(((actions, version), metrics))
 }
 
+struct MergePlanner {}
+
+#[async_trait]
+impl QueryPlanner for MergePlanner {
+    async fn create_physical_plan(
+        &self,
+        logical_plan: &LogicalPlan,
+        session_state: &SessionState,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        let planner = Arc::new(Box::new(DefaultPhysicalPlanner::with_extension_planners(
+            vec![Arc::new(MergeMetricExtensionPlanner {})],
+        )));
+        planner
+            .create_physical_plan(logical_plan, session_state)
+            .await
+    }
+}
+
```

Comment:
Eventually we should consolidate this into a `DeltaPlanner`, some early experiments are in the deltalake-sql crate, but that is not helpful right now 😆 - maybe add a comment?